### PR TITLE
Task/331 cli artifact delete command

### PR DESCRIPTION
**Pull request by fsavoia**
# Description
This task is about creating the command to delete artifacts from TUF Metadata using RSTUF API.

#331 

# Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)

# Additional requirements
- [x] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

# Code of Conduct
By submitting this PR, you agree to follow our [Code of Conduct](CODE_OF_CONDUCT.rst).

- [x] I agree to follow this project's Code of Conduct

### CODE REVIEW

File: tests/unit/helpers/test_cli.py [hunk 1]
```diff
@@ -79,7 +80,30 @@ def test_create_artifact_payload_from_filepath(
...
+        """
+
+        path = "/fake/path/"
+
+        expected_artifact_payload = {
+            "targets": [f"{path}{temp_file.split('/')[-1]}"]
+        }
+
+        assert (
+            create_artifact_delete_payload_from_filepath(
```
Comment (MVrachev) [inline tests/unit/helpers/test_cli.py:106]: Would be better and more readable if you assign a variable `result` to the `create_artifact_delete_payload_from_filepath` call and check it with an assert.

File: tests/unit/cli/artifact/test_delete.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+            artifact_path,  # artifact filepath
+            "--path",
+            path,
+            "--api-server",
+            "fake-server",
+        ]
+
+        delete.send_payload = pretend.call_recorder(lambda *a, **kw: "123")
+
+        with client.isolated_filesystem():
```
Comment (MVrachev) [inline tests/unit/cli/artifact/test_delete.py:31]: Nice!
I haven't used that, but it seems useful!

File: tests/unit/cli/artifact/test_delete.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                    "targets": [
+                        artifact_path,
+                    ],
+                },
+                expected_msg="Remove Artifact(s) successfully submitted.",
+                command_name="Artifact Deletion",
+            )
+        ]
+
+    def test_delete_without_api_server(self, client, test_context):
```
Comment (MVrachev) [inline tests/unit/cli/artifact/test_delete.py:95]: One more test will be nice 
`test_delete_not_passing_api_server_but_existing_in_settings`
where your `api_server` will be inside `settings.SERVER`.

File: tests/unit/helpers/test_cli.py [hunk 4]
```diff
@@ -79,7 +80,30 @@ def test_create_artifact_payload_from_filepath(
...
         assert (
-            create_artifact_payload_from_filepath(
+            create_artifact_add_payload_from_filepath(
+                filepath=temp_file,
+                path=path,
+            )
+            == expected_artifact_payload
+        )
+
+    def test_create_artifact_delete_payload_from_filepath(
```
Comment (MVrachev) [inline tests/unit/helpers/test_cli.py:90]: One more test will be useful:
`create_artifact_add_payload_from_filepath_without_path`

**Review by MVrachev — CHANGES_REQUESTED** [review]
Great work @fsavoia!
I left a few comments.

Additionally, we usually don't push the `png` file changes as they come from underline changes of the underline tool `plantuml`.

**Comment by fsavoia** [comment]
Thank @MVrachev  for your good suggestions. I updated the PR with the improvements.

File: docs/source/_static/repository-service-tuf-cli-C1.png
Comment (MVrachev) [inline docs/source/_static/repository-service-tuf-cli-C1.png:1]: I didn't mean you need to remove the changes in the file `docs/source/devel/repository_service_tuf.cli.artifact.rst`.
We do want the documentation for `rstuf artifact delete` as you included it.
Can you please remove this commit https://github.com/repository-service-tuf/repository-service-tuf-cli/pull/483/commits/b09d806fcaf07c17e62cb5ccd304807b07a4ec47 and instead remove those `png` file changes.
These files are the ones changed by `plantuml`.

**Review by MVrachev — CHANGES_REQUESTED** [review]
One small think is left.

File: docs/source/_static/repository-service-tuf-cli-C1.png
Comment (fsavoia) [reply docs/source/_static/repository-service-tuf-cli-C1.png:1]: Now I got it Martin.

**Comment by MVrachev** [comment]
Thanks @fsavoia for your work!
Just to let you know I meant removing the binary files:
- `docs/source/_static/repository-service-tuf-cli-C1.png`
- `docs/source/_static/repository-service-tuf-cli-C2.png`
- `docs/source/_static/repository-service-tuf-cli-C3.png`
from our changes, but this is not a big deal and we can approve it.
I wanted to let you know for future to not include them your prs.

**Review by MVrachev — APPROVED** [review]

**Review by MVrachev — APPROVED** [review]

**Pull request merged**